src/main/java/Parser.java

```diff
@@ -19,174 +19,90 @@
      * methods in the TaskList class.
      */
-    public void scan() {
-        while (true) {
-            String input = sc.nextLine();
-            ArrayList<Task> list = TaskList.getTaskList();
-            try {
-                if (input.equals("bye")) {
-                    UI.printBye();
-                    break;
-                } else if (input.contains("find")) {
-                    if (input.split(" ").length == 1) {
-                        throw new DukeException("Sorry! Please input a task to find");
-                    } else {
-                        String keyword = input.substring(5);
-                        TaskList.findTasks(keyword);
-                    }
-                } else if (input.equals("list")) {
-                    UI.printList();
-                } else if (input.contains("done")) {
-                    String[] inputArr = input.split(" ");
-                    int num = Integer.parseInt(inputArr[1]);
-                    if (num > list.size()) {
-                        throw new DukeException("Sorry! That is an invalid task number.");
-                    } else {
-                        TaskList.doneAction(num);
-                    }
-                } else if (input.contains("todo")) {
-                    if (input.split(" ").length == 1) {
-                        throw new DukeException("Sorry! Description of todo cannot be empty.");
-                    } else {
-                        String name = input.substring(5);
-                        Task todo = new ToDos(name);
-                        TaskList.addToList(todo);
-                    }
-                } else if (input.contains("deadline")) {
-                    if (input.split(" ").length == 1) {
-                        throw new DukeException("Sorry! Description of deadline cannot be empty.");
-                    } else {
-                        int index = input.indexOf(" ");
-                        String action = input.substring(index + 1);
-                        String[] actionArr = action.split(" /by ");
-                        if (index == -1) {
-                            throw new DukeException("Sorry! Please enter a deadline.");
-                        } else {
-                            String by = actionArr[1];
-                            String name = actionArr[0];
-                            Task deadline = new Deadline(name, by);
-                            TaskList.addToList(deadline);
-                        }
-                    }
-                } else if (input.contains("event")) {
-                    if (input.split(" ").length == 1) {
-                        throw new DukeException("Sorry! Description of event cannot be empty.");
+    public static Command parse(String fullCommand) throws DukeException {
+        Command command = null;
+        String[] commandSplit = fullCommand.split(" ", 2);
```

Comment:
-> split, 2, can you remember this when you look back at your codes one month later? :)